src/Microsoft.Dnx.Compilation.CSharp/RoslynProjectReference.cs

```diff
@@ -68,48 +69,69 @@ public IList<ISourceReference> GetSources()
                                      .ToList();
         }
 
-        public Assembly Load(IAssemblyLoadContext loadContext)
+        public Assembly Load(AssemblyName assemblyName, IAssemblyLoadContext loadContext)
         {
             using (var pdbStream = new MemoryStream())
             using (var assemblyStream = new MemoryStream())
             {
-                IList<ResourceDescription> resources = CompilationContext.Resources;
-
-                Logger.TraceInformation("[{0}]: Emitting assembly for {1}", GetType().Name, Name);
-
-                var sw = Stopwatch.StartNew();
+                var afterCompileContext = new AfterCompileContext
+                {
+                    ProjectContext = CompilationContext.ProjectContext,
+                    Compilation = CompilationContext.Compilation,
+                    AssemblyStream = assemblyStream,
+                    SymbolStream = pdbStream
+                };
 
                 EmitResult emitResult = null;
 
-                if (_supportsPdbGeneration.Value)
+                if (!string.IsNullOrEmpty(assemblyName.CultureName) && !assemblyName.CultureName.Equals("neutral"))
```

Comment:
Make a helper, IsNeutralResourceCulture and use it here and in the comparer. That method should have the comment about mono and the special case around neutral. Also use string.Equals.